File: C#0.7/Limit.cs
using System;

namespace MyMath.Calculus
{
	public class Limit
	{
		public Limit(Func<double, double> f, double a, double dx)
		{
			this.f = f;
			this.a = a;
			this.dx = dx;

		}
		public double a, dx;
		public Func<double, double> f;

		public double Apply()
		{
			return (f(a + dx) + f(a - dx)) / 2.0;

		}

	}

}

File: 2.1.0/JGJAR/libraries/MyMath/calculus/Limit.java
package JGJAR.libraries.MyMath.calculus;

import JGJAR.libraries.MyMath.DoubleFunction;
import JGJAR.libraries.MyMath.MyMath;

/**<h1>Limit class</h1><br>
 * 
 * @author JavaGamesJAR
 * @since MyMath 2.0.0
 * @version 1.0
 */
public class Limit 
{
	public Limit(DoubleFunction f, double a, double dx)
	{
		this.f = f;
		this.a = a;
		this.dx = dx;
		
	}
	DoubleFunction f;
	double dx, a;
	
	/**Calculates the limit
	 * 
	 * @return lim x->a f(x)
	 */
	public double apply()
	{
		return (f.execute(a-dx)+f.execute(a+dx))/2.0;
		
	}

}
